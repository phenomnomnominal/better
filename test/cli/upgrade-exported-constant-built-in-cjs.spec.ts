// eslint-disable-next-line require-extensions/require-extensions -- tests not ESM ready yet
import { createFixture } from '../fixture';

const ARGV = ['node', './bin/betterer', 'upgrade'];

describe('betterer upgrade', () => {
  it('should upgrade exported constant built-in tests in a CommonJS module', async () => {
    const { cleanup, logs, paths } = await createFixture(
      'upgrade-exported-constant-built-in-cjs',
      {
        './.betterer.ts': `
const { regexp } = require('@betterer/regexp');

module.exports.noHack = regexp(/HACK/i).include('**/*.ts');
        `
      },
      {
        logFilters: [/ðŸŒŸ Upgrading Betterer/]
      }
    );

    const fixturePath = paths.cwd;

    const { cli__ } = await import('@betterer/cli');

    await cli__(fixturePath, ARGV);

    expect(logs).toMatchSnapshot();

    await cleanup();
  });
});
