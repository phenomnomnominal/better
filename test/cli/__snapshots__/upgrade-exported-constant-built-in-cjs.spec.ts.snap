// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`betterer upgrade > should upgrade exported constant built-in tests in a CommonJS module 1`] = `
[
  "ðŸŽ‰ Upgrading Betterer (0ms): 1 task done!
âœ… Upgrading "./.betterer.js": done!
ãƒ»
ãƒ» Betterer can upgrade your test definition file automatically! âœ¨
ãƒ» Here's what the changes will look like:
ãƒ»
ãƒ» - Before
ãƒ» + After
ãƒ»
ãƒ» - const { regexp } = require('@betterer/regexp');
ãƒ» + const { regexp } = require("@betterer/regexp");
ãƒ»
ãƒ» - module.exports.noHack = regexp(/HACK/i).include('**/*.ts');
ãƒ» + module.exports.noHack = () => regexp(/HACK/i).include("**/*.ts");
ãƒ» +
ãƒ»
ãƒ» If that looks okay, run \`betterer upgrade --save\` to write the changes to disk. ðŸ’¾
ãƒ»",
]
`;
